### PR TITLE
Remove uninlinable variables

### PR DESCRIPTION
**Pull request by telephon**
We shouldn't use uninlinable variables in a quark, this fixes it.

### CODE REVIEW

File: classes/JSONlib.sc [hunk 1]
```diff
@@ -40,10 +40,10 @@ JSONlib {
 	}
 
 	prConvertToJson {|v|
-		var array;
+		var array, val;
```
Comment (capital-G) [inline classes/JSONlib.sc:43]: Can you elaborate the advantage of this? Its rather nitpicky but now the closure of the variable is bigger than it needs to be. Is this some kind of optimization?

**Review by capital-G — APPROVED** [review]

File: classes/JSONlib.sc (hunk 1 above)
Comment (telephon) [reply classes/JSONlib.sc:43]: Yes, there is a special byte code (`JumpIfFalse`) that can be used when there are no variables in the if or while clause. And you sometimes need no `TailCallReturnFromFunction`.

```supercollider
f = { var x = 1.0.rand; if(0.5.coin) { x } { 0 } };
g = {  if(0.5.coin) { var x = 1.0.rand; x } { 0 } };

f.def.dumpByteCodes
g.def.dumpByteCodes
```

Comment (capital-G) [reply classes/JSONlib.sc:43]:
```supercollider
(
var deltas = 10.collect({
	var start = Date.getDate.rawSeconds;
	var delta;
	1000000.do({f.()});
	delta = Date.getDate.rawSeconds - start;
	"took %".format(delta);
	delta;
});
// deltas.postln;
"mean\t\t%\nstdev\t%".format(deltas.mean, deltas.stdDev).postln;
)

```

`f`
```
mean	0.15460321903229
stdev	0.0037942190453099
```


`g`
```
mean	0.71153500080109
stdev	0.015307572883915
```

so around factor 5.

I am more on the (python) camp with a favor on clean code over optimization :)

> Programmers waste enormous amounts of time thinking about, or worrying about, the speed of noncritical parts of their programs, and these attempts at efficiency actually have a strong negative impact when debugging and maintenance are considered. We should forget about small efficiencies, say about 97% of the time: premature optimization is the root of all evil. Yet we should not pass up our opportunities in that critical 3%.

from donald knuth https://pic.plover.com/knuth-GOTO.pdf

Comment (telephon) [reply classes/JSONlib.sc:43]: > I am more on the (python) camp with a favor on clean code over optimization :)

This is not just the python camp, I do the same.

But the question is what is clean code! :)



- one could argue that if one's function to `if` is so large, then one should consider refactoring the code into a method call
- I am all for functions as objects, also as arguments passed to the method `if`. But as long as you get warnings from that, I have usually found a good way of rewriting it.
- there were people who wanted to work on making the inlining with inner variables, which would be the best. But I can't say if that would be possible in all cases considered.

**Pull request merged**